python/ray/data/tests/test_all_to_all.py

```diff
@@ -1771,7 +1773,7 @@ def get_node_id():
     node1_id = ray.get(get_node_id.options(resources={"bar:1": 1}).remote())
     node2_id = ray.get(get_node_id.options(resources={"bar:2": 1}).remote())
 
-    ds = ray.data.range(100, parallelism=2).random_shuffle()
+    ds = ray.data.range(100, parallelism=2).random_shuffle().materialize()
```

Comment:
do these tests not work without adding `materialize`? this seems like a regression?

Reply:
Yeah I agree, I would expect to see no test code changes for this PR.

Reply:
I think they are just for debugging. Removed them. The tests still pass.